scripts/py_matter_yamltests/matter_yamltests/parser.py

```diff
@@ -303,17 +305,38 @@ def _update_with_definition(self, container: dict, mapping_type):
 
         for value in list(container['values']):
             for key, item_value in list(value.items()):
-                mapping = mapping_type if self.is_attribute else mapping_type[value['name']]
+                if self.is_attribute:
+                    mapping = mapping_type
+                else:
+                    target_key = value['name']
+                    if mapping_type.get(target_key) is None:
+                        for candidate_key in mapping_type:
+                            if candidate_key.lower() == target_key.lower():
+                                raise KeyError(
+                                    f'"{self.label}": Unknown key: "{target_key}". Did you mean "{candidate_key}" ?')
+                        raise KeyError(
+                            f'"{self.label}": Unknown key: "{target_key}". Candidates are: "{[ key for key in mapping_type]}".')
+                    mapping = mapping_type[target_key]
 
                 if key == 'value':
                     value[key] = self._update_value_with_definition(
                         item_value, mapping)
                 elif key == 'saveAs' and type(item_value) is str and item_value not in self._parsing_config_variable_storage:
                     self._parsing_config_variable_storage[item_value] = None
                 elif key == 'constraints':
+                    # Only apply update_value_with_definition to constraints that have a value that depends on
+                    # the the value type for the target field.
+                    type_related_constraints = [
```

Comment:
Can you let me know what yaml was failing here. I am not sure I am following why this is needed at this moment

Reply:
Basically all the tests that contains a constraint with `type: int64u`. 
The code after that tries to convert  the string `int64u` to a int and obviously fails. But the bad thing is that it should not try to convert those.
For some examples:
```
src/app/tests/suites/certification/Test_TC_ACL_2_4.yaml
src/app/tests/suites/certification/Test_TC_CGEN_2_1.yaml
src/app/tests/suites/certification/Test_TC_DGETH_2_1.yaml
src/app/tests/suites/certification/Test_TC_DGGEN_2_1.yaml
src/app/tests/suites/certification/Test_TC_DGTHREAD_2_1.yaml
src/app/tests/suites/certification/Test_TC_DGWIFI_2_1.yaml
src/app/tests/suites/certification/Test_TC_MEDIAPLAYBACK_6_3.yaml
```

Reply:
Thanks... I see the problem now. How do you feel about this list coming from constraints.py itself (that way all constraint related things are contained to that one file?

Reply:
I ma starting to think that I prefer a list of constraints that do not reflect the type. That way if someone accidentally doesn't add it in when creating a new constraint they will more quickly find out why getting a keyerror instead of silently passing here For example, it seems like `startsWith` and `endsWith` should be in this list since they are strings and the type of value would be a string. Yet this could go undetected for a long time, and figuring out to add it in here is something that would take someone a little longer to figure out  versus the other way

Reply:
I have updated the code such that `constraints.py` expose a `is_typed_constraint` method that will raise an error if the constraint is unknown. That way we are sure to catch it quickly.